src/Model/Behavior/SearchBehavior.php

```diff
@@ -214,10 +214,13 @@ protected function _processFilters($filters, $params, $query)
             $filter->args($params);
             $filter->query($query);
 
-            if (!$filter->skip()) {
+            if ($filter->skip()) {
+                continue;
+            }
+            $result = $filter->process();
+            if ($result !== false) {
```

Comment:
Explicit check for `false` will cause problems with existing custom filters people have which don't return anything.